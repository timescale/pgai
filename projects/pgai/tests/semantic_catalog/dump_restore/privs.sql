-- check table privileges
select
  n.nspname as "schema"
, k.relname as "table"
, u as "user"
, p as "privilege"
, case has_table_privilege(u, k.oid, p) when true then 'YES' else 'no' end as granted
from unnest('{ada,vera,edith}'::text[]) u
inner join pg_namespace n on (n.nspname = any(array['ai', 'postgres_air']))
inner join pg_class k on (n.oid = k.relnamespace and k.relkind in ('r', 'p'))
cross join unnest(array['select', 'insert', 'update', 'delete']) p
where k.relname like 'semantic_catalog%'
order by n.nspname, k.relname, u, p
;
-- check sequence privileges
select
  n.nspname as "schema"
, k.relname as "sequence"
, u as "user"
, p as "privilege"
, case has_sequence_privilege(u, k.oid, p) when true then 'YES' else 'no' end as granted
from unnest('{ada,vera,edith}'::text[]) u
inner join pg_namespace n on (n.nspname = any(array['ai', 'postgres_air']))
inner join pg_class k on (n.oid = k.relnamespace and k.relkind in ('S'))
cross join unnest(array['select', 'update', 'usage']) p
where k.relname like 'semantic_catalog%'
order by n.nspname, k.relname, u, p
;
