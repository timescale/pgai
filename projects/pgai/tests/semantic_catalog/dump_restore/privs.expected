 schema |           table            | user  | privilege | granted 
--------+----------------------------+-------+-----------+---------
 ai     | semantic_catalog           | ada   | delete    | YES
 ai     | semantic_catalog           | ada   | insert    | YES
 ai     | semantic_catalog           | ada   | select    | YES
 ai     | semantic_catalog           | ada   | update    | YES
 ai     | semantic_catalog           | edith | delete    | no
 ai     | semantic_catalog           | edith | insert    | no
 ai     | semantic_catalog           | edith | select    | YES
 ai     | semantic_catalog           | edith | update    | no
 ai     | semantic_catalog           | vera  | delete    | no
 ai     | semantic_catalog           | vera  | insert    | no
 ai     | semantic_catalog           | vera  | select    | YES
 ai     | semantic_catalog           | vera  | update    | no
 ai     | semantic_catalog_embedding | ada   | delete    | YES
 ai     | semantic_catalog_embedding | ada   | insert    | YES
 ai     | semantic_catalog_embedding | ada   | select    | YES
 ai     | semantic_catalog_embedding | ada   | update    | YES
 ai     | semantic_catalog_embedding | edith | delete    | no
 ai     | semantic_catalog_embedding | edith | insert    | no
 ai     | semantic_catalog_embedding | edith | select    | YES
 ai     | semantic_catalog_embedding | edith | update    | no
 ai     | semantic_catalog_embedding | vera  | delete    | no
 ai     | semantic_catalog_embedding | vera  | insert    | no
 ai     | semantic_catalog_embedding | vera  | select    | YES
 ai     | semantic_catalog_embedding | vera  | update    | no
 ai     | semantic_catalog_fact_1    | ada   | delete    | YES
 ai     | semantic_catalog_fact_1    | ada   | insert    | YES
 ai     | semantic_catalog_fact_1    | ada   | select    | YES
 ai     | semantic_catalog_fact_1    | ada   | update    | YES
 ai     | semantic_catalog_fact_1    | edith | delete    | no
 ai     | semantic_catalog_fact_1    | edith | insert    | no
 ai     | semantic_catalog_fact_1    | edith | select    | YES
 ai     | semantic_catalog_fact_1    | edith | update    | no
 ai     | semantic_catalog_fact_1    | vera  | delete    | YES
 ai     | semantic_catalog_fact_1    | vera  | insert    | YES
 ai     | semantic_catalog_fact_1    | vera  | select    | YES
 ai     | semantic_catalog_fact_1    | vera  | update    | YES
 ai     | semantic_catalog_obj_1     | ada   | delete    | YES
 ai     | semantic_catalog_obj_1     | ada   | insert    | YES
 ai     | semantic_catalog_obj_1     | ada   | select    | YES
 ai     | semantic_catalog_obj_1     | ada   | update    | YES
 ai     | semantic_catalog_obj_1     | edith | delete    | no
 ai     | semantic_catalog_obj_1     | edith | insert    | no
 ai     | semantic_catalog_obj_1     | edith | select    | YES
 ai     | semantic_catalog_obj_1     | edith | update    | no
 ai     | semantic_catalog_obj_1     | vera  | delete    | YES
 ai     | semantic_catalog_obj_1     | vera  | insert    | YES
 ai     | semantic_catalog_obj_1     | vera  | select    | YES
 ai     | semantic_catalog_obj_1     | vera  | update    | YES
 ai     | semantic_catalog_sql_1     | ada   | delete    | YES
 ai     | semantic_catalog_sql_1     | ada   | insert    | YES
 ai     | semantic_catalog_sql_1     | ada   | select    | YES
 ai     | semantic_catalog_sql_1     | ada   | update    | YES
 ai     | semantic_catalog_sql_1     | edith | delete    | no
 ai     | semantic_catalog_sql_1     | edith | insert    | no
 ai     | semantic_catalog_sql_1     | edith | select    | YES
 ai     | semantic_catalog_sql_1     | edith | update    | no
 ai     | semantic_catalog_sql_1     | vera  | delete    | YES
 ai     | semantic_catalog_sql_1     | vera  | insert    | YES
 ai     | semantic_catalog_sql_1     | vera  | select    | YES
 ai     | semantic_catalog_sql_1     | vera  | update    | YES
(60 rows)

 schema |             sequence              | user  | privilege | granted 
--------+-----------------------------------+-------+-----------+---------
 ai     | semantic_catalog_embedding_id_seq | ada   | select    | YES
 ai     | semantic_catalog_embedding_id_seq | ada   | update    | YES
 ai     | semantic_catalog_embedding_id_seq | ada   | usage     | YES
 ai     | semantic_catalog_embedding_id_seq | edith | select    | no
 ai     | semantic_catalog_embedding_id_seq | edith | update    | no
 ai     | semantic_catalog_embedding_id_seq | edith | usage     | no
 ai     | semantic_catalog_embedding_id_seq | vera  | select    | no
 ai     | semantic_catalog_embedding_id_seq | vera  | update    | no
 ai     | semantic_catalog_embedding_id_seq | vera  | usage     | no
 ai     | semantic_catalog_fact_1_id_seq    | ada   | select    | YES
 ai     | semantic_catalog_fact_1_id_seq    | ada   | update    | YES
 ai     | semantic_catalog_fact_1_id_seq    | ada   | usage     | YES
 ai     | semantic_catalog_fact_1_id_seq    | edith | select    | no
 ai     | semantic_catalog_fact_1_id_seq    | edith | update    | no
 ai     | semantic_catalog_fact_1_id_seq    | edith | usage     | no
 ai     | semantic_catalog_fact_1_id_seq    | vera  | select    | YES
 ai     | semantic_catalog_fact_1_id_seq    | vera  | update    | YES
 ai     | semantic_catalog_fact_1_id_seq    | vera  | usage     | YES
 ai     | semantic_catalog_id_seq           | ada   | select    | YES
 ai     | semantic_catalog_id_seq           | ada   | update    | YES
 ai     | semantic_catalog_id_seq           | ada   | usage     | YES
 ai     | semantic_catalog_id_seq           | edith | select    | no
 ai     | semantic_catalog_id_seq           | edith | update    | no
 ai     | semantic_catalog_id_seq           | edith | usage     | no
 ai     | semantic_catalog_id_seq           | vera  | select    | no
 ai     | semantic_catalog_id_seq           | vera  | update    | no
 ai     | semantic_catalog_id_seq           | vera  | usage     | no
 ai     | semantic_catalog_obj_1_id_seq     | ada   | select    | YES
 ai     | semantic_catalog_obj_1_id_seq     | ada   | update    | YES
 ai     | semantic_catalog_obj_1_id_seq     | ada   | usage     | YES
 ai     | semantic_catalog_obj_1_id_seq     | edith | select    | no
 ai     | semantic_catalog_obj_1_id_seq     | edith | update    | no
 ai     | semantic_catalog_obj_1_id_seq     | edith | usage     | no
 ai     | semantic_catalog_obj_1_id_seq     | vera  | select    | YES
 ai     | semantic_catalog_obj_1_id_seq     | vera  | update    | YES
 ai     | semantic_catalog_obj_1_id_seq     | vera  | usage     | YES
 ai     | semantic_catalog_sql_1_id_seq     | ada   | select    | YES
 ai     | semantic_catalog_sql_1_id_seq     | ada   | update    | YES
 ai     | semantic_catalog_sql_1_id_seq     | ada   | usage     | YES
 ai     | semantic_catalog_sql_1_id_seq     | edith | select    | no
 ai     | semantic_catalog_sql_1_id_seq     | edith | update    | no
 ai     | semantic_catalog_sql_1_id_seq     | edith | usage     | no
 ai     | semantic_catalog_sql_1_id_seq     | vera  | select    | YES
 ai     | semantic_catalog_sql_1_id_seq     | vera  | update    | YES
 ai     | semantic_catalog_sql_1_id_seq     | vera  | usage     | YES
(45 rows)

